src/Shared/SqlProcessor.cs

```diff
@@ -87,62 +456,111 @@ private static bool SkipComment(string sql, ref int index)
                 }
             }
 
-            index = i;
+            state.ParsePosition = ++i;
             return true;
+#endif
         }
 
         // Scan past single-line comment
         if (ch == '-' && i + 1 < length && sql[i + 1] == '-')
         {
+#if NET
+            // Find next line break efficiently and preserve the newline for whitespace handling
+            var rest = sql.Slice(i + 2);
+            var idx = rest.IndexOfAny('\r', '\n');
+            if (idx >= 0)
+            {
+                // Position at the newline so ParseWhitespace can copy it
+                state.ParsePosition = i + 2 + idx;
+            }
+            else
+            {
+                state.ParsePosition = sql.Length;
+            }
+
+            return true;
+#else
             for (i += 2; i < length; ++i)
             {
                 ch = sql[i];
-                if (ch is '\r' or '\n')
+                if (ch == '\r' || ch == '\n')
                 {
                     i -= 1;
                     break;
                 }
             }
 
-            index = i;
+            state.ParsePosition = ++i;
             return true;
+#endif
         }
 
         return false;
     }
 
-    private static bool SanitizeStringLiteral(string sql, ref int index)
+    private static bool SanitizeStringLiteral(ReadOnlySpan<char> sql, Span<char> buffer, ref ParseState state)
     {
-        var ch = sql[index];
-        if (ch == '\'')
+        var nextChar = sql[state.ParsePosition];
+        if (nextChar == '\'')
         {
-            var i = index + 1;
+#if NET
+            var rest = sql.Slice(state.ParsePosition + 1);
+            while (!rest.IsEmpty)
+            {
+                var idx = rest.IndexOf('\'');
+                if (idx < 0)
+                {
+                    state.ParsePosition = sql.Length;
+                    return true;
+                }
+
+                if (idx + 1 < rest.Length && rest[idx + 1] == '\'')
+                {
+                    // Skip escaped quote ('')
+                    rest = rest.Slice(idx + 2);
+                    continue;
+                }
+
+                // Found terminating quote
+                state.ParsePosition = sql.Length - rest.Length + idx + 1;
+
+                buffer[state.SanitizedPosition++] = '?';
```

Comment:
Shall we make `'?'` a named constant like `RedactedChar` or similar?

Reply:
Yeah, can do.